Sprint 2/step 2

gitspace8 opened the pull request.

x121mc committed:
init sprint2/step2

x121mc committed:
1. тестова подгрузка ингов
2. создал базовый акшон/редюсер и стор
3. тестовый миделваре по мануалу с урока

x121mc committed:
1. тестова подгрузка ингов
2. создал базовый акшон/редюсер и стор
3. тестовый миделваре по мануалу с урока

x121mc committed:
1. починил HEAD

x121mc committed:
1. потерял данные где то - пересоздаю акшоны и редюсеры
2. сделал ДНД и убрал пропсы на 1м уровне

x121mc committed:
кусок переделываем левую часть блока

x121mc committed:
кусок переделываем левую часть блока - в1

x121mc committed:
кусок переделываем левую часть блока - в2 - вроде работает

x121mc committed:
кусок переделываем левую часть блока - в3 - не работает

x121mc committed:
кусок переделываем левую часть блока - в4 - фиксанул баг с источником

x121mc committed:
кусок переделываем левую часть блока - в5 вроде готово

x121mc committed:
кусок переделываем левую часть блока - в7 - убраны косяки

x121mc committed:
проверка - по чек листу

VovaGayduk42 commented on src/services/reducers/ingr-details.js:
```diff
@@ -0,0 +1,43 @@
+                selectedIngredient: null,
+            }
+        }
+        case OPEN_INGR_DETAILS_MODAL: {
```
Можно лучше:
Избегайте сокращений переменных.
Сокращения могут плохо влиять на читаемость кода.
Это очень важно, ведь в будущем ваш код будут читать другие разработчики.
Назовите переменную ingr => ingredient

gitspace8 replied:
Поправил

VovaGayduk42 commented on src/services/reducers/burger-ingredients.js:
```diff
@@ -0,0 +1,89 @@
+                ingredients: action.ingredients
+            };
+        }
+        case GET_INGREDIENTS_FAILED: {
```
Необходимо исправить:
Если при выполнении запроса к Api в усилителе произошла ошибка, то содержимое хранилища соответствующего элемента должно приводиться к начальному состоянию. Например, если пользователь оформил заказ, а при оформлении второго у него произошла ошибка — в модальном окне не должен отображаться старый номер заказа.

gitspace8 replied:
В моем понимании я очищаю ингредиенты, когда файлед

VovaGayduk42 commented on src/services/reducers/order-details.js:
```diff
@@ -0,0 +1,55 @@
+                orderId: action.orderId,
+            }
+        }
+        case PLACE_ORDER_FAILED: {
```
Необходимо исправить:
Если при выполнении запроса к Api в усилителе произошла ошибка, то содержимое хранилища соответствующего элемента должно приводиться к начальному состоянию. Например, если пользователь оформил заказ, а при оформлении второго у него произошла ошибка — в модальном окне не должен отображаться старый номер заказа.

gitspace8 replied:
В моем понимании я очищаю номер заказа, когда файлед попытка отправить заказ (смотрю это черед DevTools)
1. очищается номер заказа и ставится статус файлед.
как мне кажется.

VovaGayduk42 commented on src/services/reducers/burger-constructor.js:
```diff
@@ -0,0 +1,39 @@
+import {ADD_INGR, DELETE_INGR, MOVE_INGR, SET_BUNS,} from "../actions/burger-constructor";
```
Можно лучше:
Избегайте сокращений переменных.
Сокращения могут плохо влиять на читаемость кода.
Это очень важно, ведь в будущем ваш код будут читать другие разработчики.
Назовите переменную ingr => ingredient

gitspace8 replied:
Поправил

VovaGayduk42 commented:
Общий комментарий
Добрый день.
Отличная работа!

Сильные стороны:
Безусловно, вы разобрались в основных аспектах работы Redux и Dnd.
Функционал вашего приложения прекрасно работает!
Отдельный лайк за скролл ингридиентов по нажатию на табы =)
Так же у вас качественный код, хорошие нейминги и выбраны оптимальные подходы для реализации фич!

Необходимо исправить:

- Все изображения должны лежать в директории `images`

- Было бы правильнее не создавать несколько констант с URL для доступа к апи
  А одно, в которой будет содержаться доменное имя
  И использовать его таким образом:
  ` `${PATH}/ingredients` `

- Список ингридиентов получается от api
  Нет необходимости в моках
  Удалите их из проекта
  `/utils/data.js`

- Так же оставил комментарии в коде

Можно лучше:

- Компонент `Margin` для создания отступов - плохой подход
  Используйте стили

- Не стоит оставлять инструменты для отладки в проекте (customMiddleware)
  Для отладки Redux очень полезный инструмент ReduxToolkit
  Его должно хватать, и нет необходимости создавать customMiddleware

- Лучше не оставлять в коде неиспользуемые переменные.
  Они делают ваш код перегруженным и менее читаемым. Так же они могут негативно влиять на производительность приложения.
  Предлагаю удалить все неиспользуемые переменные.
  Их список можно увидеть в консоли сервера (Это та консоль, где вы запускаете `npm start`)

- Избегайте сокращений переменных.
  Сокращения могут плохо влиять на читаемость кода.
  Это очень важно, ведь в будущем ваш код будут читать другие разработчики.
  Назовите переменную ingr => ingredient

Прошу поправить недочеты и прислать работу на повторное ревью.
Желаю успеха!

x121mc committed:
1. поправил переменные
2. перенес картинку
3. убрал нон юзед переменную
4. свел все к 1й константе урла
5. убрал миделваре
6. удалил - тестовые данные
7. убрал сокращения

x121mc committed:
1. пропустил штуку

VovaGayduk42 commented:
Отлично! Проделана большая работа!

Про возврат состояния, можно делать так:
`return { ...initialState, failed: true }`

Принимаю)
